### ISSUE TITLE
Package NotificationCompat does not exist

### ISSUE THREAD

**Issue by hhuseyincihangir** [FrogSquare/GodotFireBase#59]
I tried so hard, but I couldn't figure it out. I've tried different versions of dependency, I haven't tried v7 and v4 versions with different combinations. I made the SDK change. I understand he doesn't see the support-v4 package. How can I solve this?

ERROR:

``` > Configure project : 
WARNING: The option setting 'android.overridePathCheck=true' is experimental and unsupported.
The current default is 'false'

NDK is missing a "toolchains" directory.
If you are using NDK, verify the ndk.dir is set to a valid NDK directory.  It is currently set to D:\sdk.
If you are not using NDK, unset the NDK variable from ANDROID_NDK_HOME or local.properties to remove this warning.


> Task :compileDebugJavaWithJavac FAILED
D:\godot-3.1.1-stable\platform\android\java\src\com\google\android\vending\expansion\downloader\impl\DownloadNotification.java:51: error: package NotificationCompat does not exist
        private NotificationCompat.Builder mActiveDownloadBuilder;
                                  ^
D:\godot-3.1.1-stable\platform\android\java\src\com\google\android\vending\expansion\downloader\impl\DownloadNotification.java:52: error: package NotificationCompat does not exist
        private NotificationCompat.Builder mBuilder;
                                  ^
D:\godot-3.1.1-stable\platform\android\java\src\com\google\android\vending\expansion\downloader\impl\DownloadNotification.java:53: error: package NotificationCompat does not exist
        private NotificationCompat.Builder mCurrentBuilder;
                                  ^
D:\godot-3.1.1-stable\platform\android\java\src\org\godotengine\godot\Godot.java:62: error: package android.support.v4.content does not exist
import android.support.v4.content.ContextCompat;
                                 ^
D:\godot-3.1.1-stable\platform\android\java\src\com\google\android\vending\expansion\downloader\impl\DownloadNotification.java:207: error: package NotificationCompat does not exist
                mActiveDownloadBuilder = new NotificationCompat.Builder(ctx);
                                                               ^
D:\godot-3.1.1-stable\platform\android\java\src\com\google\android\vending\expansion\downloader\impl\DownloadNotification.java:208: error: package NotificationCompat does not exist
                mBuilder = new NotificationCompat.Builder(ctx);
                                                 ^
D:\godot-3.1.1-stable\platform\android\java\src\com\google\android\vending\expansion\downloader\impl\DownloadNotification.java:212: error: cannot find symbol
                mActiveDownloadBuilder.setPriority(NotificationCompat.PRIORITY_LOW);
                                                   ^
  symbol:   variable NotificationCompat
  location: class DownloadNotification
D:\godot-3.1.1-stable\platform\android\java\src\com\google\android\vending\expansion\downloader\impl\DownloadNotification.java:213: error: cannot find symbol
                mActiveDownloadBuilder.setCategory(NotificationCompat.CATEGORY_PROGRESS);
                                                   ^
  symbol:   variable NotificationCompat
  location: class DownloadNotification
D:\godot-3.1.1-stable\platform\android\java\src\com\google\android\vending\expansion\downloader\impl\DownloadNotification.java:215: error: cannot find symbol
                mBuilder.setPriority(NotificationCompat.PRIORITY_LOW);
                                     ^
  symbol:   variable NotificationCompat
  location: class DownloadNotification
D:\godot-3.1.1-stable\platform\android\java\src\com\google\android\vending\expansion\downloader\impl\DownloadNotification.java:216: error: cannot find symbol
                mBuilder.setCategory(NotificationCompat.CATEGORY_PROGRESS);
                                     ^
  symbol:   variable NotificationCompat
  location: class DownloadNotification
D:\godot-3.1.1-stable\platform\android\java\src\org\godotengine\godot\Godot.java:955: error: cannot find symbol
                        if (ContextCompat.checkSelfPermission(this, Manifest.permission.RECORD_AUDIO) != PackageManager.PERMISSION_GRANTED) {
                            ^
  symbol:   variable ContextCompat
  location: class Godot
D:\godot-3.1.1-stable\platform\android\java\src\org\godotengine\godot\Godot.java:962: error: cannot find symbol
                        if (ContextCompat.checkSelfPermission(this, Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
                            ^
  symbol:   variable ContextCompat
  location: class Godot
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
12 errors


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BU¦LD FAILED in 2s
17 actionable tasks: 1 executed, 16 up-to-date

**Comment by slabgames** [comment]
use androidx package for NotificationCompat. You have to manually change missing SupportV4 classes to androidx package

For example, NotificationCompat is now using androidx.core.app.NotificationCompat

**Comment by hhuseyincihangir** [comment]
This problem is solved with the new godot 3.2.2 plugin system. Thanks for your answer.
